packages/common/src/api/tan-query/wallets/useTokenBalance.ts

```diff
@@ -50,21 +51,25 @@ export const getTokenBalanceQueryKey = (
  */
 export const useTokenBalance = ({
   mint,
+  userId: userIdParam,
   isPolling,
   pollingInterval = 1000,
   ...queryOptions
 }: {
   mint: string
+  userId?: ID
   isPolling?: boolean
   pollingInterval?: number
 } & QueryOptions) => {
   const { audiusSdk, env } = useQueryContext()
-  const { data: user } = useCurrentAccountUser()
-  const ethAddress = user?.wallet ?? null
+  const { data: user } = useUser(userIdParam, { enabled: !!userIdParam })
+  const { data: currentUser } = useCurrentAccountUser({ enabled: !userIdParam })
+  const userId = user?.user_id ?? currentUser?.user_id ?? null
+
+  const ethAddress = user?.wallet ?? currentUser?.wallet ?? null
```

Comment:
can maybe add a `user` var here in case we need to use this same user in the future. so `const userToCheck = user ?? currentUser` or something and then do `userToCheck.wallet`